File: src/main/kotlin/com/stannis/dataModel/complexStatementTypes/FinalTranslation.kt
package com.stannis.dataModel.complexStatementTypes

import com.stannis.dataModel.Statement
import com.stannis.dataModel.statementTypes.FunctionDeclarator
import com.stannis.dataModel.statementTypes.FunctionDefinition
import com.stannis.dataModel.statementTypes.LinkageSpecification

data class FinalTranslation(
    override val type: String = "FinalTranslation",
    var directives: ArrayList<String>?,
    var globalDeclaration: ArrayList<DeclarationWithParent>?,
    var internDeclaration: ArrayList<DeclarationWithParent>?,
    var methodsWithFunctionCalls: ArrayList<FunctionDefinition>?,
    var functionCallsWithoutImplementation: ArrayList<FunctionDeclarator>?,
    var classList: ArrayList<ComplexCompositeTypeSpecifier>?,
    var linkageSpecification: ArrayList<LinkageSpecification>?
    ) : Statement

File: Interpreter/Interpreter/Models/complexStatementTypes/FinalTranslation.cs
using System.Collections.Generic;

namespace Interpreter.Models
{

        public class FinalTranslation : IStatement
        {
                public IList<string> directives = new List<string>();

                public IList<DeclarationWithParent> globalDeclaration = new List<DeclarationWithParent>();

                public IList<DeclarationWithParent> internDeclaration = new List<DeclarationWithParent>();

                public IList<FunctionDefinition> methodsWithFunctionCalls = new List<FunctionDefinition>();

                public IList<FunctionDeclarator> functionCallsWithoutImplementation = new List<FunctionDeclarator>();

                public IList<ComplexCompositeTypeSpecifier> classList = new List<ComplexCompositeTypeSpecifier>();

                public IList<LinkageSpecification> linkageSpecification = new List<LinkageSpecification>();
        }

};